File: Greedy/core/StocksII.java
public class StocksII {
  public int maxProfit(int[] prices) {
        int totalProfit = 0;
        int holdPrice = prices[0];
        for(int i = 1 ; i < prices.length ; i++ ) {
            if(holdPrice > prices[i]) {
                holdPrice = prices[i];
            } else {
                totalProfit += prices[i] - holdPrice;
                holdPrice = prices[i];
            }
        }
        return totalProfit;
    }
}

File: Array/prefix-suffix/StocksII.cpp
#include<bits/stdc++.h>
using namespace std;

int maxProfit(vector<int>& prices) {
    int l=0;
    int r=1;
    int profit=0;
    while(r<prices.size())
    {
        if(prices[l]<prices[r])
        {
            profit+=prices[r]-prices[l];;
            l=r;
        }
        else
        {
            l=r;
        }
        r++;
    }
    return profit;
}

int main()
{
    return 0;
}